Src/Newtonsoft.Json/Linq/JValue.cs

```diff
@@ -350,21 +350,17 @@ internal static int Compare(JTokenType valueType, object objA, object objB)
                     }
 #endif
                 case JTokenType.Bytes:
-                    if (!(objB is byte[]))
+                    byte[] bytes2 = objB as byte[];
+                    if (bytes2 == null)
                     {
                         throw new ArgumentException("Object must be of type byte[].");
                     }
 
                     byte[] bytes1 = objA as byte[];
-                    byte[] bytes2 = objB as byte[];
                     if (bytes1 == null)
```

Comment:
I had a look at the code and I don't see any way to create a JValue with `TokenType.Bytes` that doesn't have a value of 'byte[]'.